extension/src/Documentation.ts

```diff
@@ -41,47 +58,174 @@ export async function generateExternalDocumentation() {
+            && ((<ALProcedure>proc).access === ALAccessModifier.public)
+            || (<ALProcedure>proc).event).length > 0)
+        || ([ALObjectType.Page, ALObjectType.PageExtension].includes(obj.getObjectType()) && (!obj.apiObject))
+    );
 
-    await generateObjectsDocumentation(docsRootPath, tocItems, publicObjects, removeObjectNamePrefixFromDocs, createTocSetting);
+    await generateObjectsDocumentation(docsRootPath, tocItems, publicObjects, removeObjectNamePrefixFromDocs, createTocSetting, ignoreTransUnitsSetting);
 
-    await generateWebServicesDocumentation(tocItems, createTocSetting);
-    await generateApiDocumentation(objects, tocItems);
+    let webServices = await generateWebServicesDocumentation(docsRootPath, objects, tocItems, createTocSetting);
+    let apiObjects = await generateApiDocumentation(docsRootPath, objects, tocItems);
+
+
+    if (generateDeprecatedFeaturesPageWithExternalDocsSetting) {
+        generateDeprecatedFeaturesPage(docsRootPath, objects, publicObjects, webServices, apiObjects, tocItems);
+    }
 
     if (createTocSetting) {
         let tocContent = YamlItem.arrayToString(tocItems);
         saveContentToFile(tocPath, tocContent);
     }
 
-    if (GenerateTooltipDocsWithExternalDocs) {
+    if (generateTooltipDocsWithExternalDocsSetting) {
         generateToolTipDocumentation(objects);
     }
 
-    async function generateApiDocumentation(objects: ALObject[], toc: YamlItem[]) {
-        let apiObjects = objects.filter(o => ((o.objectType === ALObjectType.Page && o.getPropertyValue(ALPropertyType.PageType)?.toLowerCase() === 'api') || (o.objectType === ALObjectType.Query && o.getPropertyValue(ALPropertyType.QueryType)?.toLowerCase() === 'api')) && o.getPropertyValue(ALPropertyType.EntityName));
+    function generateDeprecatedFeaturesPage(docsRootPath: string, objects: ALObject[], objectsWithPage: ALObject[], webServices: ALTenantWebService[], apiObjects: ALObject[], toc: YamlItem[]) {
+        const filename = "deprecated-features.md";
+        const obsoleteIndexPath = path.join(docsRootPath, filename);
+        let publicObjects = objects.filter(x => x.publicAccess && (!x.apiObject)).sort((a, b) => {
+            if (a.objectType !== b.objectType) {
+                return a.objectType.localeCompare(b.objectType);
+            }
+            return a.name.localeCompare(b.name);
+        });
+        let headerItem: YamlItem = new YamlItem({ name: 'Deprecated Features', href: filename });
+        let subItems: YamlItem[] = [];
+        headerItem.items = subItems;
+        toc.push(headerItem);
+        const header = `# Deprecated Features`;
+        let indexContent = '';
+
+        objectTypeHeaderMap.forEach((header: string, type: ALObjectType) => {
+            indexContent = generateDeprecatedTable(docsRootPath, type, header, DocsType.Public, indexContent, publicObjects, objectsWithPage, subItems);
+        });
+        let wsObjects: ALObject[] = [];
+        webServices.forEach(x => {
```

Comment:
This is just a thought, not a suggestion. But I think this could be a one liner with array map. Not a 100% on the syntax here. 
```javascript
let wsObjects: ALObject[] = webServices.map( (x) => { isNullOrUndefined(x.object) ? return: return x.object; } )
```
I just started toying with varying forEatch with Array.map and it's clear that they have their different use cases. TypeScript sometimes even tell you that "maybe your better of with a forEach" :D

Reply:
I did write it with a map() first 😁
But the result of the map() function was an array of ALObject|undefined, and that was not what I wanted... If you could solve that I would be happy to use the map() function

Reply:
Did you try adding a type declartion to the arrow function? Maybe the null check should be moved to a filter instead :)

Reply:
I tried enough to abandon the idea 😁
If you solve it somehow, please share